### PR TITLE
Added page specific JS (fix 37)

### LINKED ISSUE #37: ShopifyAppBar should not be initialized in view template
It's largely an extension of the view already, and not having to deal with that in the same place as we handle other front-end tasks like adding products would make our js easier to read.

### PR DESCRIPTION
**Pull request by jaydenwindle**
## Description

- Added auto loading of page-specific JS (e.g. when `giveaways#index` is loaded, `assets/javascripts/giveaways/index.js` gets included)
- moved dash_main.js and dash_submit.js to their respective page-specific js files

## Motivation and Context

Step in the right direction to fix #37, still need to remove the inline js in `giveaways/show.html.erb` and `giveaways/winners.html.erb`

## How Has This Been Tested?

Manual testing in dev env, all js functionality is still working

## Screenshots (if appropriate):
N/A
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [guidelines](CONTRIBUTING.md) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/assets/javascripts/dash_main.js [hunk 1]
```diff
@@ -33,3 +33,4 @@
     }
   }
 })
+;
```
Comment (karimsa) [inline app/assets/javascripts/dash_main.js:36]: Semicolons are not needed in JS.

File: app/assets/javascripts/dash_submit.js [hunk 2]
```diff
@@ -144,3 +144,4 @@
     }
   }
 })
+;
```
Comment (karimsa) [inline app/assets/javascripts/dash_submit.js:147]: Semicolons are not needed in JS.

**Review by karimsa — CHANGES_REQUESTED** [review]
Other than the semicolons, it looks good.

Was there a way to add a wildcard in the rails config for manually removing files from the bundle? That way we won't have to remember to do that every time there's new JS.

File: app/assets/javascripts/dash_main.js (hunk 1 above)
Comment (jaydenwindle) [reply app/assets/javascripts/dash_main.js:36]: Sorry, bad habit. I learned JS circa 2009 so it's largely just muscle memory :stuck_out_tongue:

**Comment by jaydenwindle** [comment]
@karimsa I tried adding wildcards but couldn't get it to work for me. I'll add the files manually for now and then try to change it to wildcards later

**Pull request merged**